docs/src/main/asciidoc/writing-native-applications-tips.adoc

```diff
@@ -0,0 +1,129 @@
+
+[source]
+----
+Caused by: org.xml.sax.SAXException: SAX2 driver class com.sun.org.apache.xerces.internal.parsers.SAXParser not found
+java.lang.ClassNotFoundException: com.sun.org.apache.xerces.internal.parsers.SAXParser
+at org.xml.sax.helpers.XMLReaderFactory.loadClass(XMLReaderFactory.java:230)
+----
+
+If one were to execute the native image generation process manually, the `-H:ReflectionConfigurationFiles=` flag would be used to point to a json configuration file that specifies the program elements that would be accessed reflectively.
+
+Quarkus however makes registration of reflection a breeze by using the `ReflectiveClassBuildItem`, thus eliminating the need for such a json configuration file.
+
+To solve the problem shown in the example above one would simply have to declare:
+
+[source,java]
+----
+    @BuildStep
+    ReflectiveClassBuildItem reflection() {
+        // we only need to register the method for reflection, not the fields thus we pass true as the second and false as the arguments
+        return new ReflectiveClassBuildItem("com.sun.org.apache.xerces.internal.parsers.SAXParser", true, false);
+    }
+----
+
+More information about reflection in GraalVM can be found https://github.com/oracle/graal/blob/master/substratevm/REFLECTION.md[here].
+
+=== Alternative with @RegisterForReflection
+
+In application code it is often needed to pass classes to libraries that use reflection to figure out the methods/fields of those classes.
+One such example is using http://json-b.net/[JSON-B] like so:
+
+[source,java]
+----
+    public class Person {
+        private String first;
+        private String last;
+
+        public String getFirst() {
+            return first;
+        }
+
+        public void setFirst(String first) {
+            this.first = first;
+        }
+
+        public String getLast() {
+            return last;
+        }
+
+        public void setValue(String last) {
+            this.last = last;
+        }
+    }
+
+    @Path("/person")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public class PersonResource {
+
+        private final Jsonb jsonb;
+
+        public PersonResource() {
+            jsonb = JsonbBuilder.create(new JsonbConfig());
+        }
+
+        @GET
+        public Response list() {
+            return Response.ok(jsonb.fromJson("{\"first\":  \"foo\", \"last\":  \"bar\"}", Person.class)).build();
+        }
+    }
+----
+
+If we were to use the code above, we would get an exception like the following when using the native image:
+
+[source]
+----
+ Exception handling request to /person: org.jboss.resteasy.spi.UnhandledException: javax.json.bind.JsonbException: Can't create instance of a class: class org.acme.jsonb.Person, No default constructor found
+----
+
+To get around the problem, simply annotate `Person` with `@RegisterForReflection` so Quarkus will register the class with GraalVM as a class using reflection.
```

Comment:
This is a mouthful: "will register the class with GraalVM as a class using reflection".

Maybe `will register the class as using reflection`?